src/bootstrap/cloud/terraform/service-account.tf

```diff
@@ -49,42 +49,20 @@ resource "google_service_account_iam_policy" "robot-service" {
   policy_data        = data.google_iam_policy.robot-service.policy_data
 }
 
-resource "google_project_iam_member" "robot-service-storage" {
-  project = data.google_project.project.project_id
-  role    = "roles/storage.objectAdmin"
-  member  = "serviceAccount:${google_service_account.robot-service.email}"
-}
-
-resource "google_project_iam_member" "robot-service-account-container-access" {
-  project = var.private_image_repositories[count.index]
-  count   = length(var.private_image_repositories)
-  role    = "roles/storage.objectViewer"
-  member  = "serviceAccount:${google_service_account.robot-service.email}"
-}
-
-# Unused by the cloud-robotics stack, but existing users might need to carefully migrate
-resource "google_project_iam_member" "robot-service-datastore" {
-  project = data.google_project.project.project_id
-  role    = "roles/datastore.user"
-  member  = "serviceAccount:${google_service_account.robot-service.email}"
-}
-
-resource "google_project_iam_member" "robot-service-monitoring" {
-  project = data.google_project.project.project_id
-  role    = "roles/monitoring.metricWriter"
-  member  = "serviceAccount:${google_service_account.robot-service.email}"
-}
-
-resource "google_project_iam_member" "robot-service-tracing" {
-  project = data.google_project.project.project_id
-  role    = "roles/cloudtrace.agent"
-  member  = "serviceAccount:${google_service_account.robot-service.email}"
-}
-
-resource "google_project_iam_member" "robot-service-logging" {
-  project = data.google_project.project.project_id
-  role    = "roles/logging.logWriter"
-  member  = "serviceAccount:${google_service_account.robot-service.email}"
+resource "google_project_iam_member" "robot-service-roles" {
+  project  = data.google_project.project.project_id
+  member   = "serviceAccount:${google_service_account.robot-service.email}"
+  for_each = toset([
+    "roles/cloudtrace.agent",      # Upload cloud traces
+    # Unused by the cloud-robotics stack, but existing users might need to carefully migrate
+    "roles/datastore.user",
+    "roles/logging.logWriter",      # Upload text logs to Cloud logging
+    # Required to use robot-service@ for GKE clusters that simulate robots
+    "roles/monitoring.viewer",      
+    "roles/storage.objectAdmin",
+    "roles/storage.objectViewer",  # GCR access
```

Comment:
`roles/storage.objectViewer` needs to be separate, it's applied to `var.private_image_repositories[count.index]` not `data.google_project.project.project_id`.